### PR TITLE
CheckUser Addon

### PR DESCRIPTION
**Pull request by SnakeAmazing**
This addon is quite similar to the actual CheckPunish addon but instead of using the punishment id to check punishment data it uses the player name and returns if its ban or muted.

As far as I know, this addon works as intented and has no known errors.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -264,7 +264,7 @@
 		<module>bans-core-addons</module>
 		<module>bans-env</module>
 		<module>bans-distribution</module>
-	</modules>
+    </modules>
```
Comment (A248) [inline pom.xml:267]:
```suggestion
	</modules>
```
Undo this change

File: bans-core-addons/command-checkuser/src/main/java/space/arim/libertybans/core/addon/checkuser/CheckUserCommand.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            if (!command().hasNext()) {
+                sender().sendMessage(config.usage());
+                return null;
+            }
+
+            String name = command().next();
+
+            return uuidManager.lookupPlayer(name)
+                    .thenCompose(optUuid -> {
+                        if (optUuid.isEmpty()) {
```
Comment (A248) [inline bans-core-addons/command-checkuser/src/main/java/space/arim/libertybans/core/addon/checkuser/CheckUserCommand.java:81]: This variable isn't an `Optional<UUID>` but rather `Optional<UUIDAndAddress>`, so please choose a more appropriate name.

**Review by A248 — COMMENTED** [review]
A few minor matters before this is ready to merge:
1. See attached comments
2. You'll need to add the checkuser module as dependency of the `addon-integration` module. The addon-integration module collects the addon jars in the same place and runs some basic tests to ensure the addons are valid.

I will need to conduct some basic manual testing if you haven't already.

**Review by A248 — COMMENTED** [review]
A few minor matters before this is ready to merge:
1. See attached comments
2. You'll need to add the checkuser module as dependency of the `addon-integration` module. The addon-integration module collects the addon jars in the same place and runs some basic tests to ensure the addons are valid.

I will need to conduct some basic manual testing if you haven't already.

**Comment by SnakeAmazing** [comment]
I don't know why all the commits have been duplicated.

Ignoring that, I've added all your comments.

**Review by KoxSosen — COMMENTED** [review]
Thank you for your work!

For me, there is just a small thing; I'm not sure if this is for a seperate pull request, or commit, but I think adding a section for the addon to the [documentation ](https://docs.libertybans.org/#/Addons) might be beneficial for the users, and for keeping track of all the addons.

If you could do that in this pull request, that would be appreciated. Thank you!

**Comment by SnakeAmazing** [comment]
> Thank you for your work!
> 
> For me, there is just a small thing; I'm not sure if this is for a seperate pull request, or commit, but I think adding a section for the addon to the [documentation ](https://docs.libertybans.org/#/Addons) might be beneficial for the users, and for keeping track of all the addons.
> 
> If you could do that in this pull request, that would be appreciated. Thank you!

Done!

**Comment by A248** [comment]
I forgot to mention to use tabs rather than spaces. It's my bad, so I reformatted the files as appropriate.

I've now updated the `CONTRIBUTING.md` file with some brief code style information.
